Difi.SikkerDigitalPost.Klient/Api/SikkerDigitalPostKlient.cs

```diff
@@ -133,66 +144,105 @@ public Transportkvittering Send(Forsendelse forsendelse, bool lagreDokumentpakke
+        /// Kvitteringer blir tilgjengeliggjort etterhvert som de er klare i meldingsformidler. Det er ikke mulig � ettersp�rre kvittering for en 
         /// spesifikk forsendelse. 
         /// </summary>
-        /// <param name="kvitteringsforespørsel"></param>
+        /// <param name="kvitteringsforesp�rsel"></param>
         /// <param name="forrigeKvittering"></param>
         /// <returns></returns>
         /// <remarks>
         /// <list type="table">
-        /// <listheader><description>Dersom det ikke er tilgjengelige kvitteringer skal det ventes følgende tidsintervaller før en ny forespørsel gjøres</description></listheader>
+        /// <listheader><description>Dersom det ikke er tilgjengelige kvitteringer skal det ventes f�lgende tidsintervaller f�r en ny foresp�rsel gj�res</description></listheader>
         /// <item><term>normal</term><description>Minimum 10 minutter</description></item>
         /// <item><term>prioritert</term><description>Minimum 1 minutt</description></item>
         /// </list>
         /// </remarks>
-        public Kvittering HentKvitteringOgBekreftForrige(Kvitteringsforespørsel kvitteringsforespørsel, Forretningskvittering forrigeKvittering)
+        public async Task<Kvittering> HentKvitteringOgBekreftForrigeAsync(Kvitteringsforesp�rsel kvitteringsforesp�rsel, Forretningskvittering forrigeKvittering)
         {
             if (forrigeKvittering != null)
             {
                 Bekreft(forrigeKvittering);
             }
 
-            Logging.Log(TraceEventType.Information, "Henter kvittering for " + kvitteringsforespørsel.Mpc);
+            Logging.Log(TraceEventType.Information, "Henter kvittering for " + kvitteringsforesp�rsel.Mpc);
 
-            var guidHandler = new GuidHandler();
-            var envelopeSettings = new EnvelopeSettings(kvitteringsforespørsel, _databehandler, guidHandler);
-            var kvitteringsenvelope = new KvitteringsforespørselEnvelope(envelopeSettings);
+            var guidHandler = new GuidUtility();
+            var envelopeSettings = new EnvelopeSettings(kvitteringsforesp�rsel, _databehandler, guidHandler);
+            var kvitteringsenvelope = new Kvitteringsforesp�rselEnvelope(envelopeSettings);
 
-            Logging.Log(TraceEventType.Verbose, "Envelope for kvitteringsforespørsel" + Environment.NewLine + kvitteringsenvelope.Xml().OuterXml);
+            Logging.Log(TraceEventType.Verbose, "Envelope for kvitteringsforesp�rsel" + Environment.NewLine + kvitteringsenvelope.Xml().OuterXml);
 
             ValiderKvitteringsEnvelope(kvitteringsenvelope);
 
-            var soapContainer = new SoapContainer { Envelope = kvitteringsenvelope, Action = "\"\"" };
-            var kvittering = SendSoapContainer(soapContainer);
+            var soapContainer = new SoapContainer(kvitteringsenvelope);
+            var kvittering = await SendSoapContainer(soapContainer);
```

Comment:
Jeg henger ikke helt med på hva du legger i "async all the way". Når jeg tar "await DoSomething" så sier jeg ikke at jeg skal vente her til det er ferdig, men gå tilbake til kallende tråd om det tar tid. Await garanterer bare at det kallet i den metoden  gjøres før resten i den metoden.

Mvh Aleksander Sjåfjell
-- Sendt fra min mobil




On Wed, Jun 17, 2015 at 3:30 AM -0700, "Kristian" <notifications@github.com<mailto:notifications@github.com>> wrote:


In Difi.SikkerDigitalPost.Klient/Api/SikkerDigitalPostKlient.cs<https://github.com/difi/sikker-digital-post-klient-dotnet/pull/93#discussion_r32613330>:

>
>              ValiderKvitteringsEnvelope(kvitteringsenvelope);
>
> -            var soapContainer = new SoapContainer { Envelope = kvitteringsenvelope, Action = "\"\"" };
> -            var kvittering = SendSoapContainer(soapContainer);
> +            var soapContainer = new SoapContainer(kvitteringsenvelope);
> +            var kvittering = await SendSoapContainer(soapContainer);


Samme her. Prøv på async all the way

—
Reply to this email directly or view it on GitHub<https://github.com/difi/sikker-digital-post-klient-dotnet/pull/93/files#r32613330>.